test/Carter.Analyzers.Tests/CarterModuleShouldNotHaveDependenciesTests.cs

```diff
@@ -0,0 +1,323 @@
+namespace Carter.Analyzers.Tests;
+
+using Microsoft.CodeAnalysis.CSharp.Testing;
+using Microsoft.CodeAnalysis.Testing;
+using Microsoft.CodeAnalysis.Testing.Verifiers;
+
+public sealed class CarterModuleShouldNotHaveDependenciesTests
+{
+    [Theory]
+    [InlineData("class")]
+    [InlineData("struct")]
+    [InlineData("record")]
+    [InlineData("record struct")]
+    public Task TypeWithDependency_Diagnostic(string type)
+    {
+        var code = $$"""
+                     using Carter;
+                     using Microsoft.AspNetCore.Routing;
+
+                     {{type}} MyCarterModule : ICarterModule
+                     {
+                         internal {|#0:MyCarterModule|}(string s) {}
```

Comment:
Out of interest what is the `\#0:` stuff?

Reply:
This is Roslyn's markup syntax, allowing to specify where a diagnostic is supposed to be emitted, without having to specify the exact line and column span at which the diagnostic occurs in the `DiagnosticResult`. Hence the `.WithLocation(0);`. If the analyzer emitted diagnostics at multiple locations, I would use {|#1: and then call `.WithLocation(1);`.